### ISSUE TITLE
Bug --cache broken if using `kit.typescript.config` in `svelte.config.js`

### ISSUE THREAD

**Issue by rebasecase** [sveltejs/eslint-plugin-svelte#1293]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.32.0

### What version of `eslint-plugin-svelte` are you using?

6.1.1

### What did you do?

Using this https://svelte.dev/docs/kit/configuration#typescript

Breaks eslint with `--cache` because functions can't be serialized.

https://stackblitz.com/edit/sveltejs-kit-template-default-uaahfr9a?file=eslint.config.js

### What did you expect to happen?

To work fine

### What actually happened?

```
Oops! Something went wrong! 🙁

ESLint: 9.32.0

The requested operation requires ESLint to serialize configuration data,
but the configuration key "typescript.config" contains a function value,
which cannot be serialized.

Please double-check your configuration for errors.
This feature breaks eslint
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://stackblitz.com/edit/sveltejs-kit-template-default-uaahfr9a?file=svelte.config.js

**Comment by rebasecase** [comment]
What I think is needed is svelte to generate a serializable version of `svelte.config`, which can be included in the eslint config?

**Comment by rebasecase** [comment]
Since there's no public api for this I tried this..

```ts
function generateSvelteConfig() {
  // Find base tsconfig.json
  const tsConfigPath = fileURLToPath(new URL('./tsconfig.json', import.meta.url));
  const baseConfigJson = typescript.readConfigFile(tsConfigPath, typescript.sys.readFile).config;

  // Expand "extends" and normalize paths
  const parsedConfig = typescript.parseJsonConfigFileContent(
    baseConfigJson,
    typescript.sys,
    process.cwd()
  );

  let tsConfigJson = parsedConfig.raw || baseConfigJson;

  // // Run the SvelteKit typescript.config hook if present
  if (typeof svelteConfig?.kit?.typescript?.config === 'function') {
    const retVal = svelteConfig.kit.typescript.config(tsConfigJson);
    if (retVal && typeof retVal === 'object') {
      tsConfigJson = retVal;
    }
  }

  // Create a serializable version of svelteConfig for ESLint caching
  // We need to execute the typescript.config function if it exists
  const finalSvelteConfig = {
    ...svelteConfig,
    kit: {
      ...svelteConfig.kit,
      typescript: {
        ...svelteConfig.kit.typescript,
        config: tsConfigJson
      }
    }
  };

  return finalSvelteConfig;
}
```

**Comment by rebasecase** [comment]
Not actually sure that's even needed, doesn't it read tsconfig anyway, so can't you delete the key?

**Comment by baseballyama** [comment]
I don’t think there’s anything that can be done on the plugin side, so this will have to be handled by the user.

```js
    languageOptions: {
      parserOptions: {
        svelteConfig: {
          ...svelteConfig,
          kit: { ...svelteConfig.kit, typescript: undefined },
        },
      },
    },
```

Since this can be easily resolved using the method above, and because there’s no solution other than handling it in userland, I’ll close this issue.

**Comment by marekdedic** [comment]
Drive-by comment: Couldn't this be handled on SvekteKit's side? Since this project is now part of the "official" svelte org, it seems like a reasonable request for kit...
